tasklist/els-schema/src/main/java/io/camunda/tasklist/util/IndexSchemaValidatorUtil.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.tasklist.util;
+
+import static io.camunda.zeebe.client.impl.Loggers.LOGGER;
+
+import com.google.common.collect.Maps;
+import io.camunda.tasklist.exceptions.TasklistRuntimeException;
+import io.camunda.tasklist.property.TasklistProperties;
+import io.camunda.tasklist.schema.IndexMapping;
+import io.camunda.tasklist.schema.IndexMapping.IndexMappingProperty;
+import io.camunda.tasklist.schema.IndexMappingDifference;
+import io.camunda.tasklist.schema.SemanticVersion;
+import io.camunda.tasklist.schema.indices.IndexDescriptor;
+import io.camunda.tasklist.schema.manager.SchemaManager;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public abstract class IndexSchemaValidatorUtil {
+
+  private static final Pattern VERSION_PATTERN = Pattern.compile(".*-(\\d+\\.\\d+\\.\\d+.*)_.*");
+
+  public static String getIndexPrefix(final TasklistProperties tasklistProperties) {
```

Comment:
if you change the static methods into class methods, you can replace this with `schemaManager.getIndexPrefix`